src/include/txn.i

```diff
@@ -514,8 +547,21 @@ __wt_txn_visible(
 static inline bool
 __wt_txn_upd_visible(WT_SESSION_IMPL *session, WT_UPDATE *upd)
 {
-	return (__wt_txn_visible(session,
-	    upd->txnid, WT_TIMESTAMP_NULL(&upd->timestamp)));
+	WT_VISIBLE_TYPE visible_type;
+
+	visible_type = __wt_txn_visible(session, upd->txnid,
+	    WT_TIMESTAMP_NULL(&upd->timestamp), upd);
```

Comment:
What I expected was for `__wt_txn_visible` to continue a boolean.

Then wait if `upd->state == LOCKED`.  Then save `upd->state`, call `__wt_txn_visible` and make sure the state doesn't change (loop if it does).